src/core/execute.c

```diff
@@ -7070,6 +7085,48 @@ int exec_directory_add(ExecDirectoryItem **d, size_t *n, const char *path, char
         return 0;
 }
 
+int exec_store_log_filter_pattern(ExecContext *c, const char *pattern, bool is_allowlist) {
+        _cleanup_(pattern_freep) pcre2_code *compiled_pattern = NULL;
+        int r;
+
+        assert(c);
+        assert(pattern);
+
+        if (isempty(pattern)) {
+                /* Empty assignment resets the lists. */
+                c->log_filter_allowed_patterns = set_free_free(c->log_filter_allowed_patterns);
+                c->log_filter_denied_patterns = set_free_free(c->log_filter_denied_patterns);
```

Comment:
`set_free()` is enough, as `set_put_strdup()` uses `string_hash_ops_free`, which free()s stored strings.